### PR TITLE
TFD-254 Add module for including Beam API.

### PR DESCRIPTION
**Pull request by RyanSkraba**
**Please check if the PR fulfills these requirements**
- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)

There are no dependencies on Beam APIs or runtime objects in the component framework.

**What is the new behavior?**

Components that depend on Beam APIs or create runtime objects should include the component-api-beam module.

**Does this PR introduce a breaking change?**
- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

There is no breaking change for Studio components -- they should ignore the two interfaces present in the `component-api-beam` package.

### CODE REVIEW

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/264/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/264/)

File: components-api-beam/src/main/java/org/talend/components/api/beam/BeamJobBuilder.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+    /**
+     * Builds the processing into the Beam job using the context provided.
+     * 
+     * @param ctx All of the information available for the job running in the current Beam
+     * {@link org.apache.beam.sdk.Pipeline}, at the time this component is added. The component is responsible for
+     * keeping the context in a known state.
+     */
+    void build(BeamJobContext ctx);
+}
```
Comment (talend-glorieux) [inline components-api-beam/src/main/java/org/talend/components/api/beam/BeamJobBuilder.java]: Missing newline at EOF

**Comment by francisu** [comment]
I was imagining components-adaptor-beam. We will have adaptor code for different environments (DI, Beam, BD/Spark-S, BD/Spark-B, BD/MapR). I would like to see these all in components-adaptor, since really they are not part of the components-api stuff.

What do you think?

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/267/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/267/)

File: components-api-beam/pom.xml [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    <version>0.15.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+
+    <name>Component API - Beam</name>
+
+    <properties>
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+
+        <maven.compiler.source>1.8</maven.compiler.source>
```
Comment (sgandon) [inline components-api-beam/pom.xml]: all components related APIs must be done in Java 7 right now.

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/295/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/295/)

File: components-filedelimited/src/main/java/org/talend/components/filedelimited/runtime/DelimitedAdaptorFactory.java [hunk 3]
```diff
@@ -19,6 +19,10 @@
```
Comment (sgandon) [inline components-filedelimited/src/main/java/org/talend/components/filedelimited/runtime/DelimitedAdaptorFactory.java]: heuuu, je pense que ces modifications n'appartiennent pas à cette PR sur beam.

**Comment by build-talend** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/308/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/308/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/311/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/311/)

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/385/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/385/)

File: core/components-adapter-beam/src/main/java/org/talend/components/api/beam/BeamJobBuilder.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+// This source code is available under agreement available at
+// %InstallDIR%\features\org.talend.rcp.branding.%PRODUCTNAME%\%PRODUCTNAME%license.txt
+//
+// You should have received a copy of the agreement
+// along with this program; if not, write to Talend SA
+// 9 rue Pages 92150 Suresnes, France
+//
+// ============================================================================
+package org.talend.components.api.beam;
+
```
Comment (francisu) [inline core/components-adapter-beam/src/main/java/org/talend/components/api/beam/BeamJobBuilder.java]: should be org.talend.components.adapters.beam, should not be in api package

**Comment by build-talend** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/512/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/512/)

**Pull request merged**